File: src/main/java/com/yoku/colovia/Repository/UserRepository.java
package com.yoku.colovia.Repository;

import com.yoku.colovia.Entity.User.UserProfile;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface UserRepository extends JpaRepository<UserProfile, Long> {
    Optional<UserProfile> findByUserId(UUID userId);
}

File: src/main/kotlin/com/yoku/colovia/repository/UserRepository.kt
package com.yoku.colovia.repository

import com.yoku.colovia.entity.user.UserProfile
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<UserProfile, UUID> {
    fun findByEmail(email: String): Optional<UserProfile>

}
